bin/improver-orographic-enhancement

```diff
@@ -0,0 +1,107 @@
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to calculate orographic enhancement."""
+
+import datetime
+import os
+import iris
+import numpy as np
+
+from improver.argparser import ArgParser
+from improver.wind_calculations.wind_components import ResolveWindComponents
+from improver.orographic_enhancement import OrographicEnhancement
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+
+
+def main():
+    """Calculate orographic enhancement of precipitation from model pressure,
+    temperature, relative humidity and wind input files"""
+
+    parser = ArgParser(description="Calculate orographic enhancement.")
+
+    parser.add_argument('temperature_filepath', metavar='TEMPERATURE_FILEPATH',
+                        help='Full path to input NetCDF temperature file')
+    parser.add_argument('humidity_filepath', metavar='HUMIDITY_FILEPATH',
+                        help='Full path to input NetCDF relhumidity file')
+    parser.add_argument('pressure_filepath', metavar='PRESSURE_FILEPATH',
+                        help='Full path to input NetCDF pressure file')
+    parser.add_argument('windspeed_filepath', metavar='WINDSPEED_FILEPATH',
+                        help='Full path to input NetCDF wind speed file')
+    parser.add_argument('winddir_filepath', metavar='WINDDIR_FILEPATH',
+                        help='Full path to input NetCDF wind direction file')
+
+    parser.add_argument('orography_filepath', metavar='OROGRAPHY_FILEPATH',
+                        help='Full path to input NetCDF high resolution (1 km)'
+                        ' orography ancillary')
+
+    parser.add_argument('output_high_resolution', metavar='OUTPUT_HIGH_RES',
+                        help='Full path to write orographic enhancement file '
+                        'on high resolution (1 km) grid')
+    parser.add_argument('output_standard_grid', metavar='OUTPUT_STANDARD',
+                        help='Full path to write orographic enhancement file '
+                        'on standard grid')
+
+    parser.add_argument('--boundary_height_metres', type=float, default=1000.,
+                        help='Model height level to extract variables for '
+                        'calculating orographic enhancement, as proxy for '
+                        'the boundary layer.')
+
+    args = parser.parse_args()
+
+    # read input variables
+    bh = args.boundary_height_metres
+    constraint = iris.Constraint(height=lambda h: bh - 0.1 < h < bh + 0.1)
```

Comment:
What happens if you don't find a height level? Does it raise a sensible error?

Reply:
It does now :)

Reply:
We could have a CLI test which raises one of these Errors.

Reply:
Agreed.

Reply:
Additionally, the boundary layer height here is defined in metres, but there is no unit check on the cube when applying this constraint, so we could ask for 100m and get 100ft.

Reply:
Added a test for unfound height levels.